manager/integration/tests/test_zone.py

```diff
@@ -0,0 +1,190 @@
+    zone2_replica_count = 0
+
+    for replica in volume.replicas:
+        replica_host_id = replica.hostId
+
+        replica_host_zone = client.by_id_node(replica_host_id).zone
+
+        if replica_host_zone == ZONE1:
+            zone1_replica_count += 1
+        elif replica_host_zone == ZONE2:
+            zone2_replica_count += 1
+
+    assert zone1_replica_count == zone2_replica_count
+
+    for i in range(randrange(3, 5)):
+        volume = client.by_id_volume(volume_name)
+
+        replica_count = len(volume.replicas)
+        assert replica_count == 2
+
+        replica_id = randrange(0, replica_count)
+
+        replica_name = volume.replicas[replica_id].name
+        replica_host_id = volume.replicas[replica_id].hostId
+
+        replica_host_zone = client.by_id_node(replica_host_id).zone
+
+        volume.replicaRemove(name=replica_name)
+
+        wait_for_volume_degraded(client, volume_name)
+
+        wait_for_volume_healthy(client, volume_name)
+
+        wait_for_volume_replica_count(client, volume_name, replica_count)
+
+        volume = client.by_id_volume(volume_name)
+
+        replica_names = map(lambda replica: replica.name, volume["replicas"])
+
+        wait_new_replica_ready(client, volume_name, replica_names)
+
+        zone1_replica_count = 0
+        zone2_replica_count = 0
+
+        for replica in volume.replicas:
+            replica_host_id = replica.hostId
+
+            replica_host_zone = client.by_id_node(replica_host_id).zone
+
+            if replica_host_zone == ZONE1:
+                zone1_replica_count += 1
+            elif replica_host_zone == ZONE2:
+                zone2_replica_count += 1
+
+        assert zone1_replica_count == zone2_replica_count
+
+    volume.updateReplicaCount(replicaCount=3)
+
+    wait_for_volume_degraded(client, volume_name)
+
+    wait_for_volume_replica_count(client, volume_name, 3)
+
+    wait_for_volume_healthy(client, volume_name)
+
+    volume = client.by_id_volume(volume_name)
+
+    lh_node_names = list(map(lambda node: node.name, client.list_node()))
+
+    for replica in volume.replicas:
+        lh_node_names.remove(replica.hostId)
+
+    assert lh_node_names == []
+
+    for i in range(randrange(3, 5)):
+        volume = client.by_id_volume(volume_name)
+
+        replica_count = len(volume.replicas)
+        assert replica_count == 3
+
+        replica_id = randrange(0, replica_count)
+
+        replica_name = volume.replicas[replica_id].name
+
+        volume.replicaRemove(name=replica_name)
+
+        wait_for_volume_degraded(client, volume_name)
+
+        wait_for_volume_healthy(client, volume_name)
+
+        wait_for_volume_replica_count(client, volume_name, replica_count)
+
+        volume = client.by_id_volume(volume_name)
+
+        lh_node_names = list(map(lambda node: node.name, client.list_node()))
+
+        for replica in volume.replicas:
+            lh_node_names.remove(replica.hostId)
+
+        assert lh_node_names == []
```

Comment:
Add one more part or one more separate test case: scale the replicas to test the soft anti-affinity feature.

Reply:
fixed